packages/core/test/ActionContext-test.ts

```diff
@@ -48,6 +48,30 @@ describe('ActionContext', () => {
       });
     });
 
+    describe('getSafe', () => {
+      beforeEach(() => {
+        context = context
+          .set(key1, 'abc')
+          .set(key3, [ true, false ]);
+      });
+
+      it('should get entries', () => {
+        expect(context.getSafe(key1)).toEqual('abc');
+        expect(() => context.getSafe(key2)).toThrow(`Context entry ${key2.name} is required but not available`);
+        expect(context.getSafe(key3)).toEqual([ true, false ]);
+      });
+
+      it('should fail during compilation for an incorrect key value', () => {
+        // @ts-expect-error
```

Comment:
Instead of `@ts-expect-error`, could we explicitly check for the error?
So something like `expect(() => context.getSafe(key1)).toThrow('expected error message...')`

This allows us to test for exact error messages as well

Reply:
We are already doing this here https://github.com/comunica/comunica/blob/37074b7ddf407210352c8754b8821cc74c5c42d9/packages/core/test/ActionContext-test.ts#L60

Reply:
Oh, my bad, glanced over it too quickly :-)